Sources/Controllers/2.0/OATargetInfoViewController.mm

```diff
@@ -448,6 +371,240 @@ - (BOOL) showNearestWiki
+    }] resume];
+}
+
+- (BOOL) isCardUnique:(OAUrlImageCard *)exampleCard
+{
+    for (OAUrlImageCard *card in _cards)
+    {
+       if (card && [card.imageHiresUrl isEqualToString:exampleCard.imageHiresUrl])
+           return NO;
+    }
+    return YES;
+}
+
+- (void) updateDispayingCardsInRow:(OARowInfo *)nearbyImagesRowInfo
+{
+    if (_cards.count == 0)
+        [_cards addObject:[[OANoImagesCard alloc] init]];
+    else if ([_cards[0] isKindOfClass:OANoImagesCard.class])
+        [_cards removeObjectAtIndex:0];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [((OACollapsableCardsView *)nearbyImagesRowInfo.collapsableView) setCards:_cards];
+    });
+}
+
+- (void) addNearbyImagesIfNeeded
+{
+    if ([Reachability reachabilityForInternetConnection].currentReachabilityStatus == NotReachable)
+        return;
+    
+    OARowInfo *nearbyImagesRowInfo = [[OARowInfo alloc] initWithKey:nil icon:[UIImage imageNamed:@"ic_custom_photo"] textPrefix:nil text:OALocalizedString(@"mapil_images_nearby") textColor:nil isText:NO needLinks:NO order:0 typeName:@"" isPhoneNumber:NO isUrl:NO];
+
+    OACollapsableCardsView *cardView = [[OACollapsableCardsView alloc] init];
+    cardView.delegate = self;
+    nearbyImagesRowInfo.collapsable = YES;
+    nearbyImagesRowInfo.collapsed = [OAAppSettings sharedManager].onlinePhotosRowCollapsed;
+    nearbyImagesRowInfo.collapsableView = cardView;
+    nearbyImagesRowInfo.collapsableView.frame = CGRectMake([OAUtilities getLeftMargin], 0, 320, 100);
+    [_rows addObject:nearbyImagesRowInfo];
+    
+    _nearbyImagesRowInfo = nearbyImagesRowInfo;
+}
+
+-(void) addWikidataImage:(NSString *)wikidataTagContent poi:(OAPOI *)poi rowInfo:(OARowInfo *)nearbyImagesRowInfo
+{
+    if ([wikidataTagContent hasPrefix:@"Q"])
+    {
+        NSString *url = [NSString stringWithFormat:@"https://www.wikidata.org/w/api.php?action=wbgetclaims&property=P18&entity=%@&format=json", wikidataTagContent];
+        NSURL *urlObj = [[NSURL alloc] initWithString:url];
+        NSURLSession *aSession = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
+        
+        [[aSession dataTaskWithURL:urlObj completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+            if (((NSHTTPURLResponse *)response).statusCode == 200) {
+                if (data) {
+                    NSError *error;
+                    NSString *safeCharsString = [[NSString alloc] initWithData:data encoding:NSISOLatin1StringEncoding];
+                    NSData *safeCharsData = [safeCharsString dataUsingEncoding:NSUTF8StringEncoding];
+                    NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:safeCharsData options:NSJSONReadingAllowFragments error:&error];
+                    if (!error)
+                    {
+                        try {
+                            NSString *imageName = jsonDict[@"claims"][@"P18"][0][@"mainsnak"][@"datavalue"][@"value"];
```

Comment:
```jsonDict[@"claims"][@"P18"][0]``` this will probably cause the crahs if the result is empty.

Reply:
I've checked with this lines for simulate bad input. It didn't crash.